packages/core-data/src/selectors.ts

```diff
@@ -275,6 +276,7 @@ export function getEntityConfig(
  * See https://github.com/WordPress/gutenberg/pull/41578 for more details.
  */
 export interface GetEntityRecord {
+	normalizeArgs: ( args: any ) => any;
```

Comment:
The `args` here are always an array of arguments passed to `getEntityRecord`. These are:

```
kind: string,
name: string,
key: EntityRecordKey,
query?: GetRecordsHttpQuery
```

I guess there's a syntax where I can extract these args to a definition and reuse them?

Reply:
I think you'll have to repeat the args, like:
```ts
normalizeArgs: ( kind: string, name: string, key: EntityRecordKey ) => [ string, string, EntityRecordKey ]
```
Also, the `normalizeArgs` field should be optional.

`EntityRecordKey` is `string | number`. To avoid further type complication, we can assume that `normalizeArgs` doesn't change types, i.e., it converts `EntityRecordKey` to the same `EntityRecordKey`. Not a strict `number`.

Reply:
OK we now have

```ts
/**
 * Arguments for EntityRecord selectors.
 */
type EntityRecordArgs =
	| [ string, string, EntityRecordKey ]
	| [ string, string, EntityRecordKey, GetRecordsHttpQuery ];
```

```ts
normalizeArgs?: ( args: EntityRecordArgs ) => EntityRecordArgs;
```